### DISCUSSION TITLE
Estimating state of charge with liionpack

### DISCUSSION THREAD

**Discussion by Sanchita-ta** [pybamm-team/liionpack#281] [Q&A]
Hello.
With pybamm, I can get the _discharge capacity_ and the nominal capacity to calculate the SOC.
however, with liionpack I am struggling to get any _discharge capacity_ or _discharge cell capacity_.
How can I calculate the SOC of the battery pack using liionpack?

Thanks in advance.

**Comment by MarcoZangrandi** [comment]
Does anyone have any news on this? I would need to plot the state of charge of the cells inside the pack over time.

**Reply by philgravy** [reply]
Did you ever figure it out? I'm needing to get SOC and SOH but am having difficulty

**Reply by MarcoZangrandi** [reply]
Hi Phil. Yes, I figured it out how to get the SOC. Simply, I calculated it a posteriori. Once you run the simulation you will obtain the cell currents array and the time array. Then you can set a loop that cycles on the cells and from an initial state of charge of every single cell you can add or remove a little piece of charge called for example delta_soc comupted as "Current in a certain time step*length of that time step/Nominal capacity". (Remember that Nominal Capacity it will be in Ah so you need to convert it in As multypling it by 3600). Surely there will be a more accurate method that I don't know (if you have an idea you can tell me) but it should work pretty well even if it could be a little approximated. I hope it will help you.

**Reply by philgravy** [reply]
Awesome, I'll look into that. It seems reasonably accurate based on the way you go about calculating it. Do you have any example code I could reference? I'm relatively new to python, but I completely get it if you'd prefer to not share. Thanks for the help tho!

**Reply by MarcoZangrandi** [reply]
I don't have references. I built it on my own. I'm new too. If you want to give me a contact I can send you the part of code.

**Reply by philgravy** [reply]
That would be great. My email is on my profile.
